pybalance/utils/balance_calculators.py

```diff
@@ -216,17 +216,17 @@ def _preprocess(self, data: Union[pd.DataFrame, MatchingData]):
         )
         return data
 
-    def _fetch_pool_features(self, candidate_populations):
-        if isinstance(candidate_populations, pd.DataFrame):
-            features = self._preprocess(candidate_populations)
+    def _fetch_features(self, subset_populations, full_population_data):
```

Comment:
i'm seeing that later in the file you are still referring to candidate_populations. either do so consistently or don't change here